util/check_dif_statuses.py

```diff
@@ -171,7 +195,7 @@ def _get_last_commit_date(self, repo_top, file_path, exts):
                 logging.error(
                     f"(date) file path ({full_file_path}) does not exist.")
                 sys.exit(1)
-        return last_dif_commit_date
+        return last_dif_commit_date.date()
```

Comment:
I'm not sure if this is an important case but wouldn't using `date()` here cause the script to miss cases where the DIF is modified first and the corresponding HW later on the same date?

Reply:
Yes, good catch. I had originally kept these as datetime objs until they were printed in the table so that this wouldn't happen.

Reply:
If the DIFs are not even begun, then the `dif_last_modified` is a string indicating "-". So I ended up choosing to do a string comparison rather than a date comparison. I changed the comparison to `>=` so that if hw and DIF were updated on the same date, we conservatively assume that the DIFs need to be revisited.

Reply:
Thinking more about this - wouldn't the existing code still suffer from this problem, given that `date()` returns a `datetime.date` obj, not a `datetime.datetime` obj, so the hours / minutes / seconds information is lost? Please correct me if I am wrong.

Reply:
I believe the original code kept "last commit dates" as datetime.datetime objs, until the table printing functions, where a comparison between datetime.datetime objs was made to format the warning coloring of the dates yellow, and then then just the dates of the datetime.datetime objs were printed (in yellow if the HW was updated since the SW). Regardless, it doesn't matter too much since as you point out, if both are updated on the same day, the SW is most likely stale anyway and should be revisited.

Reply:
D'oh! You're right. The existing code does return a `datetime` object. My bad.

Reply:
@alphan, @timothytrippel please let me know if this is ok. The other alternative is to revert this and incorporate the cases in the `print_status_table` method.

Reply:
If you can add a comment stating: "This comparison intentionally only considers dates (not times) of the last commit made to both the DIF and the HW, since if they are modified on the same day, the DIF is most likely out of sync with the HW.", then LGTM.

Reply:
I have now modified it to return a formatted string instead, providing the full date and time to fix this corner case. This allows `print_status_table` to be left as it is (i.e. kept simple) - in the final table I prune the last modified vars to 10 characters to just show the date. PTAL

Reply:
LGTM. Thanks for doing this!